tgui/packages/tgui/interfaces/PersonalCrafting.tsx

```diff
@@ -1,311 +1,922 @@
-          <Table.Cell collapsing>
-            <Button
-              icon="cog"
-              content="Craft"
-              disabled={!recipe.craftable}
-              tooltip={recipe.tool_text && 'Tools needed: ' + recipe.tool_text}
-              tooltipPosition="left"
-              onClick={() =>
-                act('make', {
-                  recipe: recipe.ref,
-                })
-              }
+    <Section>
+      <Stack>
+        <Stack.Item>
+          <Box width={'64px'} height={'64px'} mr={1}>
+            <Box
+              width={'32px'}
+              height={'32px'}
+              style={{
+                'transform': 'scale(2)',
+              }}
+              m={'16px'}
+              className={classes([
+                mode ? 'cooking32x32' : 'crafting32x32',
+                'a' + item.result,
+              ])}
             />
-          </Table.Cell>
-        </tr>
-      ))}
-    </table>
+          </Box>
+        </Stack.Item>
+        <Stack.Item grow>
+          <Stack>
+            <Stack.Item grow>
+              <Box mb={0.5} bold style={{ 'text-transform': 'capitalize' }}>
+                {item.name}
+              </Box>
+              {item.desc && <Box color={'gray'}>{item.desc}</Box>}
+              <Box style={{ 'text-transform': 'capitalize' }}>
+                {item.reqs && (
+                  <Box>
+                    <Stack my={1}>
+                      <Stack.Item grow>
+                        <Divider />
+                      </Stack.Item>
+                      <Stack.Item color={'gray'}>
+                        {mode === MODE.cooking ? 'Ingredients' : 'Materials'}
+                      </Stack.Item>
+                      <Stack.Item grow>
+                        <Divider />
+                      </Stack.Item>
+                    </Stack>
+                    {Object.keys(item.reqs).map((atom_id) => (
+                      <AtomContent
+                        key={atom_id}
+                        atom_id={atom_id}
+                        amount={item.reqs[atom_id]}
+                      />
+                    ))}
+                  </Box>
+                )}
+                {item.chem_catalysts && (
+                  <Box>
+                    <Stack my={1}>
+                      <Stack.Item grow>
+                        <Divider />
+                      </Stack.Item>
+                      <Stack.Item color={'gray'}>Catalysts</Stack.Item>
+                      <Stack.Item grow>
+                        <Divider />
+                      </Stack.Item>
+                    </Stack>
+                    {Object.keys(item.chem_catalysts).map((atom_id) => (
+                      <AtomContent
+                        key={atom_id}
+                        atom_id={atom_id}
+                        amount={item.chem_catalysts[atom_id]}
+                      />
+                    ))}
+                  </Box>
+                )}
```

Comment:
all of this is duplicated and could seriously be abstracted into its own functional component instead of copy and pasting a bunch of times.
The only things that change are the titles, and the list used to do the mapping.